src/components/Sidebar/Sidebar.jsx

```diff
@@ -41,16 +51,18 @@ return (
 
         <div className="h5">Category:</div>
         <div>
-            <label className='category-label'><input onChange={()=> {
-                dispatch({ type: "TOGGLE_CATEGORY", payload: "RunningShoes" });
-                }}
-                checked={showCategory && showCategory === "RunningShoes"} type="checkbox" />Running Shoes</label>
-            <label className='category-label'><input onChange={()=> {
-                dispatch({ type: "TOGGLE_CATEGORY", payload: "CricketShoes" });
-                }}
-                checked={showCategory && showCategory === "CricketShoes"} type="checkbox" />Cricket Shoes</label>
-            <label className='category-label'><input type="checkbox" />Indoor Shoes</label>
-            <label className='category-label'><input type="checkbox" />Gym Shoes</label>
+            <label className='category-label'><input value="RunningShoes" onChange={(event)=>
+                dispatch(categoryHandler(event))}
+                type="checkbox" checked={showCategory.some((item)=>item==="RunningShoes")}/>Running Shoes</label>
+            <label className='category-label'><input value="CricketShoes" onChange={(event)=>
+                dispatch(categoryHandler(event))}
+                type="checkbox" checked={showCategory.some((item)=>item==="CricketShoes")}/>Cricket Shoes</label>
+            <label className='category-label'><input value="IndoorShoes" onChange={(event)=>
+                dispatch(categoryHandler(event))}
+                type="checkbox" checked={showCategory.some((item)=>item==="IndoorShoes")}/>Indoor Shoes</label>
+            <label className='category-label'><input value="GymShoes" onChange={(event)=>
+                dispatch(categoryHandler(event))}
+                type="checkbox" checked={showCategory.some((item)=>item==="GymShoes")}/>Gym Shoes</label>
```

Comment:
Suggestion: you can make a separate function for checked and pass category value through it and use it here